src/main/java/duke/task/Task.java

```diff
@@ -0,0 +1,47 @@
+package duke.task;
+
+import java.io.Serializable;
+
+/**
+ * Abstract class representing a task with a description.
+ */
+public abstract class Task implements Serializable {
+
+    protected String description;
+    protected boolean isDone = false;
+
+    Task(String description) {
+        this.description = description;
+    }
+
+    /**
+     * Marks the task as completed.
+     */
+    public void mark() {
+        this.isDone = true;
+    }
+
+    /**
+     * Unmarks task i.e. mark task as not completed.
+     */
+    public void unmark() {
+        this.isDone = false;
+    }
+
+    /**
+     * Represents whether the task is completed or not.
+     * @return "X" if completed, " " if not completed.
+     */
+    public String getStatusIcon() {
+        return isDone ? "X" : " ";
+    }
+
+    public boolean descriptionContains(String keyword) {
```

Comment:
Perhaps `containsDescription` or `hasDescription` would sound better for this function?